run

```diff
@@ -394,29 +395,52 @@ make_xml(){
-            -s '//devices/disk[last()]' -t elem -n source -v ''               \
-            -a '//devices/disk[last()]/source' -t attr -n pool -v "$4"        \
-            -a '//devices/disk[last()]/source' -t attr -n volume -v "${v[0]}" \
-            -s '//devices/disk[last()]' -t elem -n target -v ''               \
-            -a '//devices/disk[last()]/target' -t attr -n bus -v $bus         \
-            -a '//devices/disk[last()]/target' -t attr -n dev -v $dev         \
-            "$xml"
 
-        if [ "$bus" == "scsi" ]; then
-            xml ed --inplace                                                      \
-                -s '//devices/disk[last()]' -t elem -n address -v ''              \
-                -a '//devices/disk[last()]/address' -t attr -n controller -v '0'  \
-                -a '//devices/disk[last()]/address' -t attr -n type -v 'drive'    \
-                -a '//devices/disk[last()]/address' -t attr -n unit -v $sidx      \
-                "$xml"
-        fi
+	if [ "$bus" != "scsi" ]; then
+                        xml ed --inplace                                                                     \
+                                -s //devices -t elem -n disk -v ''                                           \
+                                -a '//devices/disk[last()]' -t attr -n type -v volume                        \
+                                -a '//devices/disk[last()]' -t attr -n device -v disk                        \
+                                -a '//devices/disk[last()]' -t attr -n snapshot -v no                        \
+                                -s '//devices/disk[last()]' -t elem -n driver -v ''                          \
+                                -a '//devices/disk[last()]/driver' -t attr -n name -v qemu                   \
+                                -a '//devices/disk[last()]/driver' -t attr -n type -v "${v[2]}"              \
+                                -a '//devices/disk[last()]/driver' -t attr -n cache -v writeback             \
+                                -a '//devices/disk[last()]/driver' -t attr -n error_policy -v stop           \
+                                -s '//devices/disk[last()]' -t elem -n source -v ''                          \
+                                -a '//devices/disk[last()]/source' -t attr -n pool -v "$4"                   \
+                                -a '//devices/disk[last()]/source' -t attr -n volume -v "${v[0]}"            \
+                                -s '//devices/disk[last()]' -t elem -n backingStore -v ''                    \
+                                -s '//devices/disk[last()]' -t elem -n target -v ''                          \
+                                -a '//devices/disk[last()]/target' -t attr -n bus -v $bus                    \
+                                -a '//devices/disk[last()]/target' -t attr -n dev -v $dev                    \
+                                "$xml"
+	else
```

Comment:
Please add a comment to the pull request:
What's changed here?
Why is this changed?

Reply:
The xml description of the VM changes here. This is necessary so that scisi disks can be described as shared.